apm-sniffer/apm-sdk-plugin/servicecomb-plugin/servicecomb-java-chassis-2.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/servicecomb/v2/TransportClientHandlerInterceptor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.servicecomb.v2;
+
+import java.lang.reflect.Method;
+import java.net.URI;
+
+import org.apache.servicecomb.core.Invocation;
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+public class TransportClientHandlerInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+        MethodInterceptResult result) throws Throwable {
+        Invocation invocation = (Invocation) allArguments[0];
+        if (!checkRegisterStatus(invocation)) {
```

Comment:
Why doesn't register means `return`?

Reply:
The logic here is that if the registration is successful, the system returns the information directly. If the registration is not successful, the system continues to print the error information.

Reply:
Should not the plugin collect the error and report the error status for this exit span? If you `return` here, it looks like the service is not calling.

Reply:
It seems that you only want to handle the error that the downstream microservice is not registered. Do you want to use this method only when there is an error or will it be called anyway?

Reply:
My point is, you are intercepting this method in ServiceComb, it means RPC is going to happen now, then an exit span should be created one way or another. `Not register` is a case of error status. I am not sure whether this fits ServiceComb, but most other RPC frameworks act in this way.

Reply:
I feel this English translation will make our discussion very difficult, if you don't mind I can add you QQ to continue the discussion, I can add you directly in the Skywalking QQ group.

Reply:
If you have to, the key issue is others reading this will lost the context.

Reply:
Maybe so. If the downstream microservice is not registered, the error is irrelevant to Skywalking. The framework has its own logic processing. Maybe 1.x was written for that reason.

Reply:
I don't know, if you are a member of ServiceComb committer team(can't confirm from GitHub), you could make the call. Or someone from ServiceComb confirms this is reasonable. But in most plugins(not servicecomb), the error should report before interceptor point.

Anyway, this is not a block from a reviewer's perspective. This is a comment from my experiences.

Reply:
I understand that you think the registry should be deleted, right? If so, I'll discuss it with them this afternoon.

Reply:
Take your time.